sortinghat/core/jobs.py

```diff
@@ -168,25 +167,24 @@ def recommend_affiliations(ctx, uuids=None, last_modified=MIN_PERIOD_DATE):
     job_ctx = SortingHatContext(ctx.user, job.id, ctx.tenant)
 
     # Create an empty transaction to log which job
-    # will generate the enroll transactions.
+    # will generate the 'enroll' transactions.
     trxl = TransactionsLog.open('recommend_affiliations', job_ctx)
 
-    for chunk in _iter_split(uuids, size=MAX_CHUNK_SIZE):
-        for rec in engine.recommend('affiliation', chunk):
-            results[rec.key] = rec.options
+    for rec in engine.recommend('affiliation', uuids, last_modified):
+        results[rec.key] = rec.options
+        try:
+            individual = find_individual_by_uuid(rec.key)
+        except NotFoundError:
+            logger.warning(f"Job {job.id} 'Individual {rec.key} not found'")
+            continue
```

Comment:
I think we don't need this part. We have the `individual.mk` in `rec.mk`, and we can create the `AffiliationRecommendation` object using `rec.mk`, as we do in `recommend_matches()` with the following code:
```
try:
    with transaction.atomic():
        AffiliationRecommendation.objects.create(individual_id=rec.mk, organization=org)
except IntegrityError:
    pass
```

We can also use `rec.key` because `recommend_affiliations()` yields the same value for both.

Reply:
I think we can only use `rec.mk` because if the list of `uuids` is not empty it will mean to create recommendations for those uuids only. In that case `key` is the `uuid` and not the `mk`